documentation/design/user-stories/shared-infra.adoc

```diff
@@ -0,0 +1,599 @@
+    name: myplan
+    namespace: planspace
+
+  conditions:
+  - type: Scheduled
+    status: "True"
+    lastTransitionTime: 2020-......
+```
+
+---
+=== CRUDLW a messaging endpoint in a namespace where messaging is enabled
+=== Managing certificates for a messaging endpoint
+
+Create a `MessagingEndpoint` resource with one of the supported types.
+
+* Supported types are `cluster`, `route`, `ingress`, `loadBalancer` or `nodePort`.
+* An endpoint can only be of a single type.
+* Support protocols are AMQP, AMQPS (AMQP over TLS), AMQP-WS and AMQP-WSS (AMQP over secure websocket).
+* For "Ingress" and "Route", only the AMQPS or AMQP-WSS maybe specified (not both!)
+* Default protocols are defined in a `MessagingPlan` for a namespace. If the plan does not specify defaults, they will be:
+** For `route` and `ingress`: AMQPS
+** For `cluster`, `nodePort` and `loadBalancer`: AMQP,AMQPS
+
+==== Resource definition
+
+Example resource with all configuration knobs exposed for all types:
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  protocols:
+  - AMQP
+  - AMQPS
+  - AMQP-WS
+  - AMQP-WSS
+
+  # Optional: Override generated hostname for endpoint. Clients must have the AMQP OPEN hostname set to this value.
+  host: example.com
+
+  # Optional: specify TLS configuration for endpoint.
+  tls:
+    # Selfsigned is default unless specified by a `MessagingPlan`.
+    selfsigned: {}
+    # OpenShift requests a service certificate signed by OpenShift service CA
+    openshift: {}
+    # External allows providing certificate externally. CN/SANs are validated against host
+    external:
+      cert: # Reuse structure from address space connector spec
+      key: # Reuse structure from address space connector spec
+
+  # Cluster type creates a on-cluster load balancer
+  cluster: {}
+
+  # NodePort type creates a nodePort service, exposing the application on an ephemeral port on all nodes.
+  nodePort: {}
+
+  # LoadBalancer type creates an external load balancer for this endpoint.
+  loadBalancer:
+    # Allow setting annotations on the service object that are specific to LoadBalancer implementations.
+    annotations:
+      aws-specific-annotation: value
+
+  # Ingress type creates an Ingress resource
+  ingress:
+    # Optional: Allow configuring annotations specific to ingress controller implementations
+    annotations: {}
+
+  # Route type maps to OpenShift Route resource
+  route:
+    # Allows controlling the TLS termination of this route. Only supported for AMQP-WSS protocol.
+    tlsTermination: passthrough | reencrypt
+
+status:
+  phase: Pending | Configuring | Active | Failed | Terminating
+  message: "Useful error message"
+
+  # Host set to auto-generated value
+  host: messaging.example.com
+
+  ports:
+  - name: endpoint1-amqp
+    port: 5672
+    protocol: AMQP
+  conditions:
+  - type: FoundTenant
+    status: "True"
+    lastTransitionTime: 2020-......
+  tls:
```

Comment:
the TLS section should include the certificate's start and end date.  This will make it easy to display expiry info in the Console and gives a convenient location for a mechanism to monitor for certificate expiry.